xrmocap/data/dataset/fourd_dataset.py

```diff
@@ -0,0 +1,386 @@
+            keypoints3d_path = os.path.join(self.meta_path,
+                                            f'scene_{scene_idx}',
+                                            'keypoints3d_GT.npz')
+            keypoints3d = Keypoints.fromfile(keypoints3d_path)
+            keypoints3d.logger = self.logger
+            keypoints3d = keypoints3d.to_tensor()
+            if self.gt_kps3d_convention is not None:
+                keypoints3d = convert_keypoints(
+                    keypoints=keypoints3d,
+                    dst=self.gt_kps3d_convention,
+                    approximate=True)
+            # save mask info into confidence
+            kps3d_mask = keypoints3d.get_mask()
+            kps3d = keypoints3d.get_keypoints()
+            kps3d[..., -1] = kps3d[..., -1] * kps3d_mask
+            kps3d[..., :3] = self.factor * kps3d[..., :3]
+            keypoints3d.set_keypoints(kps3d)
+            mscene_list.append(keypoints3d)
+        self.gt3d = mscene_list
+
+    def load_perception_2d(self):
+        """Load multi-scene keypoints2d and paf."""
+
+        mscene_keypoints_list = []
+        for scene_idx in range(self.n_scene):
+
+            pkl_names = glob.glob(
+                os.path.join(self.meta_path, f'scene_{scene_idx}', "kps2d_paf", '*pkl'))
+            n_view = len(pkl_names)
+            mview_kps2d = []
+            for view_idx in range(n_view):
+
+                img_size = (self.fisheye_params[scene_idx][view_idx].width, self.fisheye_params[scene_idx][view_idx].height)
+
+                file_name = os.path.join(self.meta_path, f'scene_{scene_idx}', "kps2d_paf", str(view_idx)+'.pkl')
+                f = open(file_name,'rb')
+                detections = pickle.load(f)
+                
+                frame_size = len(detections)
+
+                joint_size = 25
+                paf_size =26
+
+                convert_joint_size = 18
+                convert_paf_size = 19
+
+                convert_detections = []
+                for i in range(frame_size):
+                    var = {'joints':{j:[] for j in range(convert_joint_size)}, 'pafs':{k:[] for k in range(convert_paf_size)}}
+                    convert_detections.append(var)
+
+                for frame_id in range(frame_size):
+                    for joint_id in range(joint_size):
+                        #resize
+                        detections[frame_id]['joints'][joint_id][:,0] = detections[frame_id]['joints'][joint_id][:,0]*(img_size[0] - 1)
+                        detections[frame_id]['joints'][joint_id][:,1] = detections[frame_id]['joints'][joint_id][:,1] * (img_size[1] - 1)
+
+                    # for paf_id in range(paf_size):
+                    #     detections[frame_id]['pafs'][paf_id] = np.power(detections[frame_id]['pafs'][paf_id], 0.8)
+
+                    #convert keypoint type body25-skel19
+                    joint_mapping = [4, 1, 5, 11, 15, 6, 12, 16, 0, 2, 7, 13, 3, 8, 14, -1, -1, 9, 10, 17, -1, -1, 18, -1, -1]
```

Comment:
Shall we do the mapping by `xrmocap.transform.convention.keypoints_convention`? It is much more easier to read.

Reply:
The convertion of bottom-up data seems different to the top-down data. Moreover, the `paf` data convertion is not defined.